File: syncer/model/facility.js
const Sequelize = require('sequelize');
const sequelize = require('../config/pg')
var Facility = sequelize.define("facility", {
    id: {
        type: Sequelize.BIGINT,
        primaryKey: true,
    },
    name: {
        type: Sequelize.TEXT,
    },
    type: {
        type: Sequelize.TEXT,
    },
    eosUsername: {
        type: Sequelize.TEXT,
        field: 'eos_username',
    },
    publicKey: {
        type: Sequelize.TEXT,
        field: 'public_key',
    },
    email: {
        type: Sequelize.TEXT,
    },
    phoneNumber: {
        type: Sequelize.TEXT,
        field: 'phone_number',
    },
    location: {
        type: Sequelize.TEXT,
    },
    website: {
        type: Sequelize.TEXT,
    },
})

module.exports = Facility

File: BE/org_service/model/facility.go
package model

import "time"

type Facility struct {
	ID          string       `json:"id,omitempty"`
	Name        string       `json:"name,omitempty"`
	Type        FacilityType `json:"type,omitempty"`
	EosUsername string       `json:"eos_username,omitempty"`
	Email       string       `json:"email,omitempty"`
	PhoneNumber string       `json:"phone_number,omitempty"`
	Location    string       `json:"location,omitempty"`
	Website     string       `json:"website,omitempty"`
	CreatedAt   time.Time    `json:"created_at,omitempty"`
	UpdatedAt   time.Time    `json:"updated_at,omitempty"`
	DeletedAt   *time.Time   `json:"deleted_at,omitempty"`
}

// FacilityType http://apps.gs1.org/GDD/Pages/clDetails.aspx?semanticURN=urn:gs1:gdd:cl:PartyRoleCode&release=3
type FacilityType string

const (
	GROWER                 FacilityType = "GROWER"
	MANUFACTURER_OF_GOODS  FacilityType = "MANUFACTURER_OF_GOODS"
	POINT_OF_SALE          FacilityType = "POINT_OF_SALE"
	TRANSPORTATION_CARRIER FacilityType = "TRANSPORTATION_CARRIER"
	WAREHOUSE_KEEPER       FacilityType = "WAREHOUSE_KEEPER"
)

// Valid .
func (f FacilityType) Valid() bool {
	if f == GROWER ||
		f == MANUFACTURER_OF_GOODS ||
		f == POINT_OF_SALE ||
		f == TRANSPORTATION_CARRIER ||
		f == WAREHOUSE_KEEPER {
		return true
	}

	return false
}
